src/malcolm/malcolmSocketHandler.js

```diff
@@ -21,30 +21,87 @@ const configureMalcolmSocketHandlers = (inputSocketContainer, store) => {
   };
 
   socketContainer.socket.onmessage = event => {
+    console.log(event.data);
     const data = JSON.parse(event.data);
     if (data.typeid === 'malcolm:core/Delta:1.0') {
-      const changes = data.changes[0][1];
+      const { changes } = data;
       const originalRequest = store
         .getState()
         .malcolm.messagesInFlight.find(m => m.id === data.id);
 
-      switch (changes.typeid) {
+      // Messy Bits
+      const attributePath = originalRequest.path;
+      const blockName = attributePath[0];
+      const attributeName = attributePath[1];
+      let attribute;
+
+      // Pull attribute from store (if exists)
+      if (
```

Comment:
I'm not sure what's going on here, don't you just want to check if it's defined with `if store.getState().malcolm.blocks.blockName && store.getState().malcolm.blocks[blockName].attributes`? I don't really know enough about what is going on here to comment properly.

Reply:
That's indeed the aim; however the client was throwing an error and crashing if blockName or attributes didn't exist; hasOwnProperty avoids the error since it doesn't try to access the property, but the only way I managed to get it work was by the nested if's to check if blockName existed before proceeding to see if it had any attributes, and then  searching for our specific attribute. Just doing the attribute search and wrapping it in a try/catch would also work but not sure if this is good practice?